### PR TITLE
circulation: fix create new loan request

### PR DESCRIPTION
**Pull request by zzacharo**
The ui wasn't sending the `document_pid` when it was requesting a new loan for an item and thus the indexer was assuming that was there. I made it optional when the indexer tries to index the document after the loan is changed and also tried to extract the document from the item when the `create_loan` api was being called.

### CODE REVIEW

File: invenio_app_ils/circulation/utils.py [hunk 1]
```diff
@@ -38,7 +38,9 @@ def circulation_document_retriever(item_pid):
     """Retrieve items given a document."""
```
Comment (topless) [inline invenio_app_ils/circulation/utils.py]: The description does not match with what it actually returns

**Review by topless — CHANGES_REQUESTED** [review]

**Review by kpsherva — DISMISSED** [review]

File: invenio_app_ils/circulation/utils.py [hunk 2]
```diff
@@ -35,10 +35,12 @@ def circulation_items_retriever(document_pid):
 
 
 def circulation_document_retriever(item_pid):
-    """Retrieve items given a document."""
+    """Retrieves document given an item."""
     item = Item.get_record_by_pid(item_pid)
     item = item.replace_refs()
-    return item["document"]["document_pid"]
+    if item and item.get("document") and item["document"].get("document_pid"):
```
[the remark this answers is not in the document]
Comment (zzacharo) [reply invenio_app_ils/circulation/utils.py]: true I will change it ;)

**Review by topless — APPROVED** [review]

File: invenio_app_ils/circulation/api.py [hunk 3]
```diff
@@ -51,6 +53,10 @@ def create_loan(params):
         raise CirculationException(
             "Patron has already a request or active loan on this item.")
 
+    if "document_pid" not in params:
+        document_pid = circulation_document_retriever(params["item_pid"])
+        if document_pid:
+            params["document_pid"] = document_pid
```
Comment (ntarocco) [inline invenio_app_ils/circulation/api.py:59]: should we actually fail if `None`?

**Review by ntarocco — APPROVED** [review]
Thank you for fixing this!

**Pull request merged**